### DISCUSSION TITLE
Understanding dipole_pp.mod

### DISCUSSION THREAD

**Discussion by katduecker** [jonescompneurolab/hnn-core#959] [General]
Hi all, 
I'm trying to better understand how we calculate the dipole in HNN. I think I get it conceptually,  but could someone explain to me the difference between Qsum and Qtotal in dipole_pp, please?
I'm also confused about self.dpl_ref in _insert_dipole in cell.py: https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/cell.py#L691

self.dpl_ref seems to be equal to dpl_vec._ref_x[0]. Is that the size of the Dipole at location x[0]?

Thanks a bunch for your help!
Katharina

**Comment by ntolley** [comment]
Think this comment is relevant:
https://github.com/jonescompneurolab/hnn-core/blob/fbe1981ff41d22db8ab2cbc303f63bc5f306d24b/hnn_core/mod/dipole_pp.mod#L5

Seems that Qsum/Qtotal are used to handle dipole calculations when simulations are divided across cores

I would guess the same applied for dpl_ref and dpl_vec but would need to dig deeper

**Comment by jasmainak** [comment]
it's all a bit confusing because the python code is simply assigning pointers to the underlying C/C++ variables defined in the mod files. The summation is actually [happening in the mod file](https://github.com/jonescompneurolab/hnn-core/blob/fbe1981ff41d22db8ab2cbc303f63bc5f306d24b/hnn_core/mod/dipole_pp.mod#L48-L49). When you say `_ref_Qsum` you are actually referring to the `Qsum` variable in the mod file
